test/regress_finalize.c

```diff
@@ -290,6 +291,62 @@ test_fin_within_cb(void *arg)
 	;
 }
 
+static void
+timer_callback_3(evutil_socket_t fd, short what, void *arg)
+{
+	// nop
+}
+
+static void
+event_finalize_callback_3(struct event *ev, void *arg)
+{
+	struct event_and_count *evc = arg;
+
+	if (evc->count == 0) {
+		memset(evc->ev, 0, sizeof(*evc->ev));
+		event_assign(evc->ev, evc->base, -1, EV_TIMEOUT, timer_callback_3, evc);
+		evc->count += 1;
+	} else {
+		free(evc->ev);
+		free(evc);
+	}
+}
+
+static void
+test_fin_debug_use_after_free(void *arg)
+{
+	struct basic_test_data *data = arg;
+	struct event_base *base = data->base;
+
+	// Allocate ev outside of libevent
+	struct event_and_count *evc = malloc(sizeof(*evc));
+	memset(evc, 0, sizeof(*evc));
+	evc->ev = malloc(sizeof(*evc->ev));
+	memset(evc->ev, 0, sizeof(*evc->ev));
+	evc->base = base;
+
+	// Assign, add event, and finalize
+	event_assign(evc->ev, base, -1, EV_TIMEOUT, timer_callback_3, evc);
+	struct timeval tv = { .tv_sec = 4, .tv_usec = 0 };
```

Comment:
Move the variable declaration to the top of the block. Otherwise, the compiler will give a "mixed declaration and code" warning.

Reply:
Done